apps/server/src/modules/oauth-provider/controller/oauth-provider.controller.ts

```diff
@@ -102,30 +108,44 @@ export class OauthProviderController {
 	}
 
 	@Get('loginRequest/:challenge')
-	getLoginRequest(@Param() params: ChallengeParams) {
-		throw new NotImplementedException();
+	async getLoginRequest(@Param() params: ChallengeParams): Promise<LoginResponse> {
+		const loginResponse: ProviderLoginResponse = await this.oauthProviderLoginFlowUc.getLoginRequest(params.challenge);
+		const mapped: LoginResponse = this.oauthProviderResponseMapper.mapLoginResponse(loginResponse);
+		return mapped;
 	}
 
 	@Authenticate('jwt')
 	@Patch('loginRequest/:challenge')
-	patchLoginRequest(@Param() params: ChallengeParams, @Query() query: AcceptQuery, @Body() body: LoginRequestBody) {
-		throw new NotImplementedException();
+	async patchLoginRequest(
+		@Param() params: ChallengeParams,
+		@Query() query: AcceptQuery,
+		@Body() body: LoginRequestBody,
+		@CurrentUser() currentUser: ICurrentUser
+	): Promise<RedirectResponse> {
+		const redirectResponse: ProviderRedirectResponse = await this.oauthProviderLoginFlowUc.patchLoginRequest(
+			currentUser.userId,
+			params.challenge,
+			body,
+			query
+		);
+		const mapped: RedirectResponse = this.oauthProviderResponseMapper.mapRedirectResponse(redirectResponse);
+		return mapped;
 	}
 
 	@Authenticate('jwt')
 	@Patch('logoutRequest/:challenge')
 	async acceptLogoutRequest(@Param() params: ChallengeParams, @Body() body: RedirectBody): Promise<RedirectResponse> {
```

Comment:
unused param?
or should this actually be used? sounds kinda important...

Reply:
i misread the hydra doc. there is no body.